src/backend/distributed/master/worker_node_manager.c

```diff
@@ -414,6 +430,30 @@ ReferenceTablePlacementNodeList(LOCKMODE lockMode)
 }
 
 
+/*
+ * CoordinatorNode returns the WorkerNode object for coordinator node if it is
+ * added to pg_dist_node, otherwise errors out.
+ */
+WorkerNode *
+CoordinatorNode()
+{
+	if (!CoordinatorAddedAsWorkerNode())
```

Comment:
Minor: Maybe add a method such as `EnsureCoordinatorAddedAsWorkerNode` so that we can use that function whenever we need to make sure that coordinator is added as a worker node.

Reply:
Currently we use `CoordinatorAddedAsWorkerNode` in two places and they have their error messages emphasizing different aspects of not having coordinator added as worker node. To me, In the current structure, it makes more sense not to implement such an helper function.